app/src/test/java/com/mstudio/superheroarch/presentation/CharacterDetailViewModelTest.kt

```diff
@@ -0,0 +1,86 @@
+package com.mstudio.superheroarch.presentation
+
+import com.mstudio.superheroarch.RickAndMortyRepositoryInstruments
+import com.mstudio.superheroarch.presentation.detail.CharacterDetailViewModel
+import com.mstudio.superheroarch.presentation.detail.CharacterDetailViewTranslator
+import com.mstudio.superheroarch.presentation.network.NetworkManagerImpl
+import com.mstudio.superheroarch.repository.model.toCharacterData
+import com.mstudio.superheroarch.usecase.GetCharacterAndEpisodeUseCase
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+import org.mockito.kotlin.verify
+
+@ExperimentalCoroutinesApi
+class CharacterDetailViewModelTest {
+
+    private lateinit var viewModel: CharacterDetailViewModel
+    private lateinit var view: CharacterDetailViewTranslator
+    private lateinit var useCase: GetCharacterAndEpisodeUseCase
+    private val testDispatcher = UnconfinedTestDispatcher()
+    private lateinit var networkManager: NetworkManagerImpl
+
+    @Before
+    fun before() {
+        Dispatchers.setMain(testDispatcher)
+        networkManager = mock()
+        useCase = mock()
+        view = mock()
+        `when`(networkManager.hasInternetConnection()).thenReturn(true)
+
+        viewModel = CharacterDetailViewModel(view, useCase, testDispatcher, networkManager)
+    }
+
+    @Test
+    fun `given a character detail screen, when screen is visited, show all the character info`() = runTest {
+        val characterData = RickAndMortyRepositoryInstruments.givenACharacterEntity().toCharacterData()
+
+        `when`(
+            useCase.getCharacterAndEpisode(
+                characterData
+            )
+        ).thenReturn(RickAndMortyRepositoryInstruments.givenCharacterCompleteDetailData())
+        viewModel.onCharacterReceived(characterData)
+
+        verify(view).showEpisode(RickAndMortyRepositoryInstruments.givenCharacterCompleteDetailData())
+        verify(view).showEpisodeExtraData("https://image.tmdb.org/t/p/w500image", 7.0)
```

Comment:
After moving the addition of the URL to the remote entity mapper, you can now use a more simple string as image URL in instruments and also in here.